actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -56,17 +60,14 @@ module Redirecting
     # To terminate the execution of the function immediately after the +redirect_to+, use return.
     #   redirect_to post_url(@post) and return
     def redirect_to(options = {}, response_options = {})
-      raise ActionControllerError.new("Cannot redirect to nil!") unless options
-      raise AbstractController::DoubleRenderError if response_body
+      response_options[:allow_other_host] ||= _allow_other_host unless response_options.key?(:allow_other_host)
```

Comment:
Okay, good point. How about this?

```suggestion
      response_options.delete(:allow_other_host) if other_hosts_forbidden
```

If other hosts are forbidden, perhaps we should get the configuration out of params entirely, lest someone try to use it later on.

Reply:
The difficulty here is mostly that we can't break existing behaviour for Rails apps that depend on the current behaviour. Previously in #35018 I took a more extreme approach with no opt-in and it had to be reverted because it blocked upgrades. We need to add open redirect protection, but optionally turn it on when the opt-in is activated. We can make the behaviour default once we've given apps enough time to upgrade.

Reply:
Right, yes, same page. In my mind, `other_hosts_forbidden` is checking to see if the configuration has been explicitly set to disallow other hosts. Another way of writing the same thing would be this:
```ruby
response_options.delete(:allow_other_host) if _allow_other_host == false
```